app/main/lib/shared_models/audio_model.py

```diff
@@ -176,7 +176,7 @@ def get_by_doc_id_or_url(self, task):
             audios = db.session.query(Audio).filter(Audio.doc_id==task.get("doc_id")).all()
             if audios and not audio:
                 audio = audios[0]
-        elif task.get('url'):
+        if task.get('url'):
```

Comment:
Looks good. We should document somewhere that if both a `doc_id` and `url` are supplied, the URL takes priority and the hash associated with the URL will be taken. 
(Or if we don't want that, we should change this to  `if audio==None and  task.get('url'):` )
I'm good with either, but I think we should document.